pkg/operator/controller/gatewayapi/controller_test.go

```diff
@@ -0,0 +1,342 @@
+		expectDelete    []client.Object
+		expectStartCtrl bool
+	}{
+		{
+			name:            "featuregates.config.openshift.io/cluster doesn't exist",
+			existingObjects: []runtime.Object{},
+			expectCreate:    []client.Object{},
+			expectUpdate:    []client.Object{},
+			expectDelete:    []client.Object{},
+			expectStartCtrl: false,
+		},
+		{
+			name: "featuregates.config.openshift.io/cluster specifies CustomNoUpgrade but doesn't specify GatewayAPI",
+			existingObjects: []runtime.Object{
+				&configv1.FeatureGate{
+					ObjectMeta: metav1.ObjectMeta{Name: "cluster"},
+					Spec: configv1.FeatureGateSpec{
+						FeatureGateSelection: configv1.FeatureGateSelection{
+							FeatureSet: configv1.CustomNoUpgrade,
+							CustomNoUpgrade: &configv1.CustomFeatureGates{
+								Enabled: []string{"Foo", "Bar", "Baz"},
+							},
+						},
+					},
+				},
+			},
+			expectCreate:    []client.Object{},
+			expectUpdate:    []client.Object{},
+			expectDelete:    []client.Object{},
+			expectStartCtrl: false,
+		},
+		{
+			name: "featuregates.config.openshift.io/cluster specifies CustomNoUpgrade but disables GatewayAPI",
+			existingObjects: []runtime.Object{
+				&configv1.FeatureGate{
+					ObjectMeta: metav1.ObjectMeta{Name: "cluster"},
+					Spec: configv1.FeatureGateSpec{
+						FeatureGateSelection: configv1.FeatureGateSelection{
+							FeatureSet: configv1.CustomNoUpgrade,
+							CustomNoUpgrade: &configv1.CustomFeatureGates{
+								Disabled: []string{"GatewayAPI"},
+							},
+						},
+					},
+				},
+			},
+			expectCreate:    []client.Object{},
+			expectUpdate:    []client.Object{},
+			expectDelete:    []client.Object{},
+			expectStartCtrl: false,
+		},
+		{
+			name: "featuregates.config.openshift.io/cluster specifies CustomNoUpgrade and enables GatewayAPI",
+			existingObjects: []runtime.Object{
+				&configv1.FeatureGate{
+					ObjectMeta: metav1.ObjectMeta{Name: "cluster"},
+					Spec: configv1.FeatureGateSpec{
+						FeatureGateSelection: configv1.FeatureGateSelection{
+							FeatureSet: configv1.CustomNoUpgrade,
+							CustomNoUpgrade: &configv1.CustomFeatureGates{
+								Enabled: []string{"GatewayAPI"},
+							},
+						},
+					},
+				},
+			},
+			expectCreate: []client.Object{
+				crd("gatewayclasses.gateway.networking.k8s.io"),
+				crd("gateways.gateway.networking.k8s.io"),
+				crd("httproutes.gateway.networking.k8s.io"),
+			},
+			expectUpdate:    []client.Object{},
+			expectDelete:    []client.Object{},
+			expectStartCtrl: true,
+		},
+		{
+			name: "featuregates.config.openshift.io/cluster specifies TechPreviewNoUpgrade",
+			existingObjects: []runtime.Object{
+				&configv1.FeatureGate{
+					ObjectMeta: metav1.ObjectMeta{Name: "cluster"},
+					Spec: configv1.FeatureGateSpec{
+						FeatureGateSelection: configv1.FeatureGateSelection{
+							FeatureSet: configv1.TechPreviewNoUpgrade,
+						},
+					},
+				},
+			},
+			// TODO Update expectCreate and expectStartCtrl when
```

Comment:
Not any more.

Reply:
I would say "not yet"; eventually this feature will be tech preview, and then "GatewayAPI" may be added to the "TechPreviewNoUpgrade" feature set.